### PR TITLE
feat: add delete upload to w3 and API

### PR DESCRIPTION
**Pull request by rafaelramalho19**
Wanted to learn a bit more about the w3 & API integration, so I made it possible to delete uploads through the API & W3 CLI

- Closes https://github.com/web3-storage/web3.storage/issues/386

### CODE REVIEW

File: packages/w3/index.js
Comment (vasco-santos) [inline packages/w3/index.js:201]:
```suggestion
 * @param {string} cid the root CID to delete
```

File: packages/w3/index.js
Comment (vasco-santos) [inline packages/w3/index.js:202]:
```suggestion
 * @param {object} [opts]
```

let's make opts optional

File: packages/w3/index.js
Comment (vasco-santos) [inline packages/w3/index.js:207]:
```suggestion
export async function deleteUpload (cid, opts = {}) {
```

File: packages/client/src/lib.js
Comment (vasco-santos) [inline packages/client/src/lib.js:236]: humm, wondering if we should fail in case of no delete found. what do you think?

**Review by vasco-santos — CHANGES_REQUESTED** [review]

File: packages/client/src/lib.js
Comment (rafaelramalho19) [reply packages/client/src/lib.js:235]: It gracefully fails, as it says:
image

Comment (vasco-santos) [reply packages/client/src/lib.js:235]: yes, if fails gracefully in the CLI, but I am concerned on the client UX. What if we go with an approach like JS built in Map [return value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete#return_value)? We would return `true` if deleted, `false` otherwise. This would fit well with our `client.get` behaviour.

Comment (rafaelramalho19) [reply packages/client/src/lib.js:235]: Gotcha, changing it

File: packages/w3/index.js
Comment (rafaelramalho19) [reply packages/w3/index.js:202]: Any reason for that? If so, we should change it on all the commands, no?

File: packages/api/src/index.js
Comment (olizilla) [inline packages/api/src/index.js:34]: This is a blocker here. Until we get scoped api tokens as per #314 we deliberately made it so tokens can only be used for non-destructive actions like uploading and listing.

**Review by olizilla — CHANGES_REQUESTED** [review]
Nice PR! Alas, it is blocked on #314 as we do not want api tokens to be able to perform destructive actions until users are able to specify the powers a token should have when creating them.

You can't delete with an api token until we land that feature.

File: packages/w3/index.js
Comment (vasco-santos) [reply packages/w3/index.js:202]: well, given no one will use this as a lib, I will not expect problems. But typescript compiler would fail if this function is used as `x.deleteUpload(cid)` because `opts` is not documented as optional. However, none of the Object properties are required. Not a big deal here

**Pull request closed without merging**
